akka-cluster/src/test/scala/akka/cluster/GossipSpec.scala

```diff
@@ -379,6 +382,35 @@ class GossipSpec extends WordSpec with Matchers {
       merged1.version.versions.keys should not contain (VectorClock.Node(vclockName(dc2d1.uniqueAddress)))
     }
 
+    "correctly prune vector clocks based on tombstones when merging" in {
+      val gdc1 = Gossip(members = SortedSet(dc1a1, dc1b1, dc2c1, dc2d1))
+        .:+(VectorClock.Node(vclockName(dc1a1.uniqueAddress)))
+        .:+(VectorClock.Node(vclockName(dc1b1.uniqueAddress)))
+        .:+(VectorClock.Node(vclockName(dc2c1.uniqueAddress)))
+        .:+(VectorClock.Node(vclockName(dc2d1.uniqueAddress)))
+        .remove(dc1b1.uniqueAddress, System.currentTimeMillis())
+
+      gdc1.version.versions.keySet should not contain (VectorClock.Node(vclockName(dc1b1.uniqueAddress)))
+
+      val gdc2 = Gossip(members = SortedSet(dc1a1, dc1b1, dc2c1, dc2d1))
+        .seen(dc1a1.uniqueAddress)
+        .:+(VectorClock.Node(vclockName(dc1a1.uniqueAddress)))
+        .:+(VectorClock.Node(vclockName(dc1b1.uniqueAddress)))
+        .:+(VectorClock.Node(vclockName(dc2c1.uniqueAddress)))
+        .:+(VectorClock.Node(vclockName(dc2d1.uniqueAddress)))
+        .remove(dc2c1.uniqueAddress, System.currentTimeMillis())
+
+      gdc2.version.versions.keySet should not contain (VectorClock.Node(vclockName(dc2c1.uniqueAddress)))
+
+      // when we merge the two, the nodes should not be reintroduced
+      val merged1 = gdc2 merge gdc1
+      merged1.members should ===(SortedSet(dc1a1, dc2d1))
+
+      merged1.version.versions.keySet should ===(Set(
+        VectorClock.Node(vclockName(dc1a1.uniqueAddress)),
+        VectorClock.Node(vclockName(dc2d1.uniqueAddress))))
+    }
```

Comment:
Nice that this can be tested 'from the outside' without referring to tombstones (after all an implementation detail) at all.